### PR TITLE
Replace `UTF8Encoding(false)` with `Encoding.Default` part 2

### PR DESCRIPTION
**Pull request by xtqqczze**
Follow-up: #18144.

cc: @iSazonov, @daxian-dbw, @JamesWTruher

Context: dotnet/runtime#45541, #5080

### CODE REVIEW

File: src/System.Management.Automation/utils/ClrFacade.cs [hunk 1]
```diff
@@ -100,23 +100,6 @@ private static IEnumerable<Assembly> GetPSVisibleAssemblies()
...
 
-        /// <summary>
-        /// Facade for getting default encoding.
-        /// </summary>
-        internal static Encoding GetDefaultEncoding()
-        {
-            if (s_defaultEncoding == null)
-            {
-                // load all available encodings
-                EncodingRegisterProvider();
```
Comment (iSazonov) [inline src/System.Management.Automation/utils/ClrFacade.cs:111]: I don't remember if we have tests for that.

**Review by iSazonov — COMMENTED** [review]
Do we not have concerns that the .Net default will be changed in the future while we don't want such a change?

**Comment by xtqqczze** [comment]
> Do we not have concerns that the .Net default will be changed in the future while we don't want such a change?

Since .NET Core 2.0, `Encoding.Default` returns UTF-8 without BOM across all platforms. .NET will not make another breaking change like this without good reason.

In the unlikely event of a breaking change to `Encoding.Default`, I do not foresee any difficulty updating PowerShell codebase.

**Comment by xtqqczze** [comment]
@iSazonov You may be interested in the .NET breaking change to proposal to _`Encoding.UTF8`_, see dotnet/runtime#51353.

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Pull request merged**